buildtest/utils/shell.py

```diff
@@ -0,0 +1,67 @@
+import shutil
+from buildtest.exceptions import BuildTestError
+
+
+class Shell:
+    def __init__(self, shell):
+        """ The Shell initializer takes an input shell and shell options and split
+            string by shell name and options.
+
+            Parameters:
+
+            :param shell: specify shell program and any options passed to shell
+            :type shell: str
+        """
+
+        # enforce input argument 'shell' to be a string
+        if not isinstance(shell, str):
+            raise BuildTestError(
+                f"Invalid type for input: {shell} must be of type 'str'"
+            )
+
+        self.name = shell.split()[0]
+        self.opts = " ".join(shell.split()[1:])
+
```

Comment:
The function opts would immediately override the class variable of the same name.

Reply:
so what are you suggesting we do? Remove ``opts`` method?

Reply:
It would be appropriate to parse original opts in the init, but then have this function more properly named based on what it does, e.g., shell.set_opts()